editors/code/src/config.ts

```diff
@@ -128,23 +148,47 @@ export class Config {
 
         if (!prebuiltBinaryName) return null;
 
+        return this.createGithubReleaseSource(
+            prebuiltBinaryName,
+            this.extensionReleaseTag
+        );
+    }
+
+    private createGithubReleaseSource(file: string, tag: string): ArtifactSource.GithubRelease {
         return {
             type: ArtifactSource.Type.GithubRelease,
+            file,
+            tag,
             dir: this.ctx.globalStoragePath,
-            file: prebuiltBinaryName,
-            storage: this.ctx.globalState,
-            tag: Config.extensionVersion,
-            askBeforeDownload: this.cfg.get("updates.askBeforeDownload") as boolean,
             repo: {
                 name: "rust-analyzer",
                 owner: "rust-analyzer",
             }
         };
     }
 
+    get nightlyVsixSource(): ArtifactSource.GithubRelease {
+        return this.createGithubReleaseSource("rust-analyzer.vsix", NIGHTLY_TAG);
+    }
+
+    readonly installedNightlyExtensionReleaseDate = new DateStorage(
```

Comment:
We can get the date of nightly extension (.vsix, not the server binary) from version in package.json. We set version to date:

https://github.com/rust-analyzer/rust-analyzer/blob/34bc0f4f55a9c2239d77b550b9bbd61b4eef6bb1/xtask/src/dist.rs#L30-L36

Reply:
Yeah, but that date contains only the day, but no hours, minutes...
This means we will need to keep the invariant of issuing releases not more often than 1 time per day and some more trickery with 25 hours heuristics I guess